Donation-platform/src/main/webapp/addcharity.html

```diff
@@ -0,0 +1,53 @@
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+  <meta charset="utf-8" />
+  <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.3/css/select2.css" />
+  <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/select2-bootstrap-css/1.4.6/select2-bootstrap.css" />
+  <link rel="stylesheet" type="text/css" href="https://cdnjs.cloudflare.com/ajax/libs/select2/4.0.3/css/select2.css" />
+  <link rel="stylesheet" href="addCharity.css" />
+  <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.5.0/css/bootstrap.min.css" />
+  <title></title>
+</head>
+<body>
+  <!-- NAVIGATION BAR -->
+  <nav class="navbar navbar-expand-sm bg-dark navbar-dark fixed-top">
+    <a class="navbar-brand" href="index.html">Charities4U</a>
+    <ul class="navbar-nav">
+      <li class="nav-item">
+        <a class="nav-link" href="trending.html">Trending Charities</a>
+      </li>
+      <li class="nav-item">
+        <a class="nav-link" href="addCharity.html">Add Charities</a>
+      </li>
+    </ul>
+  </nav>
+  <!-- ADD CHARITY FORM -->
+  <div class="container center_div">
+    <form id="charity" action="/addCharity" method="post" name="charity">
+      <h3>Add Charity Form</h3>
+      <fieldset>
+        <input name="name" id="name" placeholder="Charity Name" type="text" tabindex="1" required="" autofocus="" />
+      </fieldset>
+      <fieldset>
+        <input name="link" id="link" placeholder="Charity Website Link" type="url" required="" />
```

Comment:
similarly add validation here to check the string is a link

Reply:
The validation is already there as is. If you attempt to add a string that is not a link it will not submit the form.